packages/react-hooks/src/useMergedRefs.ts

```diff
@@ -1,22 +1,28 @@
-import { useCallback, Ref, MutableRefObject } from 'react';
+import * as React from 'react';
+
 /**
  * React hook to merge multiple React refs (either MutableRefObjects or ref callbacks) into a single ref callback that
  * updates all provided refs
  * @param refs- Refs to collectively update with one ref value.
  */
-export function useMergedRefs<T>(...refs: (Ref<T> | undefined)[]): (instance: T) => void {
-  return useCallback(
+export function useMergedRefs<T>(...refs: (React.Ref<T> | undefined)[]): ((value: T) => void) & React.RefObject<T> {
+  const mergedCallback = (React.useCallback(
```

Comment:
I think you can get rid of some of the casts if you do `const mergedCallback: ReturnType<typeof useMergedRefs<T>> = React.useCallback...` (or some other way of getting the type)

Reply:
resolved